libs/community/langchain_community/vectorstores/sqlserver.py

```diff
@@ -0,0 +1,195 @@
+                `langchain.embeddings.base.Embeddings` interface.
+            table_name: The name of the table to use for storing embeddings.
+
+        """
+
+        self.connection_string = connection_string
+        self.embedding_function = embedding_function
+        self.table_name = table_name
+        self._bind = connection if connection else self._create_engine()
+        self.EmbeddingStore = self._get_embedding_store(table_name)
+        self._create_table_if_not_exists()
+
+    def _create_engine(self) -> Engine:
+        return create_engine(url=self.connection_string, echo=True)
+
+    def _create_table_if_not_exists(self) -> None:
+        logging.info("Creating table %s", self.table_name)
+        with Session(bind=self._bind) as session:
+            Base.metadata.create_all(session.get_bind())
+
+    def _get_embedding_store(self, name: str) -> Any:
+        global _embedding_store
+        if _embedding_store is not None:
+            return _embedding_store
+
+        class EmbeddingStore(Base):
+            """This is the base model for SQL vector store."""
+
+            __tablename__ = name
+            id = Column(Uuid, primary_key=True, default=uuid.uuid4)
+            custom_id = Column(VARCHAR, nullable=True)  # column for user defined ids.
+            query_metadata = Column(JSON, nullable=True)
+            query = Column(NVARCHAR("max"), nullable=False)
+            embeddings = Column(VARBINARY, nullable=False)
+
+        _embedding_store = EmbeddingStore
+        return _embedding_store
+
+    @property
+    def embeddings(self) -> Embeddings:
+        return self.embedding_function
+
+    @classmethod
+    def from_texts(
+        cls: type[VST],
+        texts: List[str],
+        embedding: Embeddings,
+        metadatas: List[dict] | None = None,
+        **kwargs: Any,
+    ) -> VST:
+        """Return VectorStore initialized from texts and embeddings."""
+        return super().from_texts(texts, embedding, metadatas, **kwargs)
+
+    def similarity_search(
+        self, query: str, k: int = 4, **kwargs: Any
+    ) -> List[Document]:
+        return super().similarity_search(query, k, **kwargs)
+
+    def add_texts(
+        self,
+        texts: Iterable[str],
+        metadatas: Optional[List[dict]] = None,
+        ids: Optional[List[str]] = None,
+        **kwargs: Any,
+    ) -> List[str]:
+        """Compute the embeddings for the input texts and store embeddings
+            in the vectorstore.
+
+        Args:
+            texts: Iterable of strings to add into the vectorstore.
+            metadatas: List of metadatas (python dicts) associated with the input texts.
+            ids: List of IDs for the input texts.
+            kwargs: vectorstore specific parameters.
+
+        Returns:
+            List of IDs generated from adding the texts into the vectorstore.
+        """
+
+        # Embed the texts passed in.
+        embedded_texts = self.embedding_function.embed_documents(list(texts))
```

Comment:
We also have a Document data type, which is created from text and metadata. Is this the Document data type or the texts that are passed to the embed_documents function? Since the name of the function is embed_documents, I am confused about the input type being, text or document.

Reply:
The naming might be confusing since this function is from a different class (`Embeddings`). However, `embed_documents` here is used to embed a list of texts and has nothing to do with the `Document` class. There is also an `embed_query` function which in its case takes in a single text.